spec/features/collaboration/collab_spec.rb

```diff
@@ -0,0 +1,169 @@
+      stub_request(:get, "https://api.flat.io/v2/me").to_return(status: 200,
+                                                                body: json_user_resp,
+                                                                headers: {})
+      json_score_resp = File.read('spec/fixtures/flat/user_scores.json')
+      stub_request(:get, "https://api.flat.io/v2/users/me/scores").to_return(status: 200,
+                                                                             body: json_score_resp,
+                                                                             headers: {})
+      json_score_show_resp = File.read('spec/fixtures/flat/score_show.json')
+      expected = nil
+      File.open('spec/fixtures/flat/user_scores.json') do |file|
+        file.each_line { |line| expected = JSON.parse(line, symbolize_names: true) }
+      end
+      stub_request(:get, "https://api.flat.io/v2/scores/#{expected[0][:id]}").to_return(status: 200,
+                                                                                        body: json_score_show_resp,
+                                                                                        headers: {})
+
+      json_collab_resp = File.read('spec/fixtures/flat/add_collaborator.json')
+      stub_request(:post, "https://api.flat.io/v2/scores/5ecdacce7b6e796344939fed/collaborators").to_return(status: 200,
+                                                                                                            body: json_collab_resp,
+                                                                                                            headers: {})
+      visit users_dashboard_index_path
+      within('.scores') { click_link 'Funk' }
+    end
+
+    it 'I can see the collaborators on the score' do
+      expect(current_path).to eq(scores_path)
+      within('.collaborators') do
+        expect(page.all("li").count).to eq(1)
+        expect(page).to have_content('tylerpporter')
+      end
+    end
+
+    it 'I see a button to request to collaborate if it is not my score' do
+      within('.collaborators') do
+        within ('.requests') do
+          expect(page).to have_button('Request to collaborate on this score')
+        end
+      end
+    end
+
+    it 'I do not see a button if I am already a collaborator' do
+      @user.update(username: 'tylerpporter')
+      visit users_dashboard_index_path
+      within('.scores') { click_link 'Funk' }
+
+      within('.collaborators') do
+        within ('.requests') do
+          expect(page).to_not have_button('Request to collaborate on this score')
+        end
+      end
+    end
+
+    it 'I can request to collaborate & that request shows up for the owner' do
+      collab_name = @user[:username]
+      within('.collaborators') { click_button('Request to collaborate on this score') }
+
+
+      expect(current_path).to eq(scores_path)
+
+      visit users_dashboard_index_path
+      @user.update(username: 'tylerpporter')
```

Comment:
In this test file I'm changing the user name to essentially fake logging out & in as different users. We can come back & make this a bit more robust after user login is built.